proc/api.thrift

```diff
@@ -0,0 +1,40 @@
+service Proc {
+  
+  exception InvalidRequest {
+    1: optional string message
+  }
+
+  exception ScootTooBusy {
+    1: optional i64 retryAfterMs
+  }
+  
+  enum JobType {
+    IRON_TESTS=1,
+    DEFAULT=2,
+  }
+
+  struct Command {
+    1: list<string> cmds
+  }
+
+  struct Task {
+    1: optional string id
+    2: required Command command,
+    3: optional string snapshotId,
+    4: optional string contextId
+  }
+
+  struct Job {
+    1: required string id,
+    2: required list<Task> tasks
+  }
+
+  Job RunJob(
+    1: optional string contextId,
+    2: optional JobType jobType,
+    3: required list<Task> tasks
```

Comment:
That makes sense, yet there is a reason Google proto3 removed 'required'/'optional', because they were more problematic than useful. I'm not sure what else went with that proto change though, so the same arguments may not apply directly to thrift.